### PR TITLE
Add cosine_warmup_schedule

### PR DESCRIPTION
**Pull request by guarin**
### Changes

* Add `cosine_warmup_schedule` function

### How was it tested?

The change is covered by the tests for `CosineWarmupScheduler`. I'll make a follow-up PR to refactor those tests.

### CODE REVIEW

File: lightly/utils/scheduler.py [hunk 1]
```diff
@@ -67,6 +66,76 @@ def cosine_schedule(
...
+        warmup_end_value:
+            Target value for warmup. Defaults to start_value.
+        period:
+            The number of steps over which the cosine function completes a full cycle.
+            Defaults to max_steps - warmup_steps.
+    Returns:
+        Cosine decay value.
+    """
+    if warmup_steps < 0:
+        raise ValueError("Warmup steps can't be negative")
```
Comment (philippmwirth) [inline lightly/utils/scheduler.py:105]: Minor: 
```suggestion
        raise ValueError(f"Warmup steps ({warmup_steps}) can't be negative")
```

And you can add it to the other error messages below too

File: lightly/utils/scheduler.py [hunk 2]
```diff
@@ -67,6 +66,76 @@ def cosine_schedule(
...
+    if warmup_steps > max_steps:
+        raise ValueError("Warmup steps must be <= max_steps")
+    if step > max_steps:
+        warnings.warn(
```
Comment (philippmwirth) [inline lightly/utils/scheduler.py:109]: We should add a logger...

File: lightly/utils/scheduler.py [hunk 3]
```diff
@@ -67,6 +66,76 @@ def cosine_schedule(
...
+            category=RuntimeWarning,
+        )
+
+    if warmup_end_value is None:
+        warmup_end_value = start_value
+
+    if step < warmup_steps:
+        return (
+            warmup_start_value
+            + (warmup_end_value - warmup_start_value) * (step + 1) / warmup_steps
```
Comment (philippmwirth) [inline lightly/utils/scheduler.py:120]: Can you add a comment why `step + 1` is used?

File: lightly/utils/scheduler.py [hunk 4]
```diff
@@ -67,6 +66,76 @@ def cosine_schedule(
...
+        )
+    elif period is not None:
+        return cosine_schedule(
```
Comment (philippmwirth) [inline lightly/utils/scheduler.py:123]: We can make this a bit more concise:
```python
else:
    # TODO: Explain why it's set to 1 if the period is not None.
    max_steps =  max_steps - warmup_steps if period is None else 1
    return cosine_schedule(
        step=step - warmup_steps,
        max_steps=max_steps,
        start_value=start_value,
        end_value=end_value,
        period=period,
    )
```

File: lightly/utils/scheduler.py (hunk 2 above)
Comment (philippmwirth) [reply lightly/utils/scheduler.py:109]: Not in this PR of course 😄

Comment (guarin) [reply lightly/utils/scheduler.py:109]: I think warnings is fine for library code. They can be captured by any user defined logger.

**Review by philippmwirth — APPROVED** [review]

**Pull request merged**